godtools/App/Features/Languages/ChooseLanguage/ChooseLanguageViewModel.swift

```diff
@@ -34,39 +44,43 @@ class ChooseLanguageViewModel: NSObject, ChooseLanguageViewModelType {
     let selectedLanguageIndex: ObservableValue<Int?> = ObservableValue(value: nil)
     
-    required init(flowDelegate: FlowDelegate, dataDownloader: InitialDataDownloader, languagesRepository: LanguagesRepository, languageSettingsService: LanguageSettingsService, downloadedLanguagesCache: DownloadedLanguagesCache, localizationServices: LocalizationServices, getTranslatedLanguageUseCase: GetTranslatedLanguageUseCase, analytics: AnalyticsContainer, chooseLanguageType: ChooseLanguageType) {
+    required init(flowDelegate: FlowDelegate, getSettingsPrimaryLanguageUseCase: GetSettingsPrimaryLanguageUseCase, getSettingsParallelLanguageUseCase: GetSettingsParallelLanguageUseCase,  userDidSetSettingsPrimaryLanguageUseCase: UserDidSetSettingsPrimaryLanguageUseCase, userDidSetSettingsParallelLanguageUseCase: UserDidSetSettingsParallelLanguageUseCase, userDidDeleteParallelLanguageUseCase: UserDidDeleteParallelLanguageUseCase, getLanguagesListUseCase: GetLanguagesListUseCase, downloadedLanguagesCache: DownloadedLanguagesCache, localizationServices: LocalizationServices, analytics: AnalyticsContainer, chooseLanguageType: ChooseLanguageType) {
         
         self.flowDelegate = flowDelegate
-        self.dataDownloader = dataDownloader
-        self.languagesRepository = languagesRepository
-        self.languageSettingsService = languageSettingsService
+        self.getSettingsPrimaryLanguageUseCase = getSettingsPrimaryLanguageUseCase
+        self.getSettingsParallelLanguageUseCase = getSettingsParallelLanguageUseCase
+        self.userDidSetSettingsPrimaryLanguageUseCase = userDidSetSettingsPrimaryLanguageUseCase
+        self.userDidSetSettingsParallelLanguageUseCase = userDidSetSettingsParallelLanguageUseCase
+        self.userDidDeleteParallelLanguageUseCase = userDidDeleteParallelLanguageUseCase
+        self.getLanguagesListUseCase = getLanguagesListUseCase
         self.downloadedLanguagesCache = downloadedLanguagesCache
         self.localizationServices = localizationServices
-        self.getTranslatedLanguageUseCase = getTranslatedLanguageUseCase
         self.analytics = analytics
         self.chooseLanguageType = chooseLanguageType
         
         deleteLanguageButtonTitle = localizationServices.stringForMainBundle(key: "clear")
         closeKeyboardTitle = localizationServices.stringForMainBundle(key: "dismiss")
-        
-        super.init()
-                
+                        
         switch chooseLanguageType {
         case .primary:
             navTitle.accept(value: localizationServices.stringForMainBundle(key: "primary_language"))
         case .parallel:
             navTitle.accept(value: localizationServices.stringForMainBundle(key: "parallel_language"))
         }
         
-        reloadData()
-        
-        setupBinding()
+        Publishers.Zip3(getLanguagesListUseCase.getLanguagesList(), getSettingsPrimaryLanguageUseCase.getPrimaryLanguage(), getSettingsParallelLanguageUseCase.getParallelLanguage())
```

Comment:
Does this need to be `CombineLatest` then, too?  Seems like it should be. Can you combine more than 2 streams?

Reply:
Hey @rachaelblue this should also use CombineLatest.   I will include this in GT-1746